File: Compiler/Code/bench/multi_lang_benchmark/w1_fibonacci/Fib.java
public class Fib {
    public static long fib(long n) {
        if (n < 2) return n;
        return fib(n - 1) + fib(n - 2);
    }

    public static void main(String[] args) {
        long t0 = System.nanoTime();
        long res = fib(30);
        long t1 = System.nanoTime();
        long us = (t1 - t0) / 1000;
        System.out.println("W1_FIB checksum=" + res + " time_us=" + us);
    }
}

File: Compiler/Code/bench/multi_lang_benchmark/w1_fibonacci/fib.py
import time

def fib(n: int) -> int:
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)

def main():
    t0 = time.perf_counter()
    res = fib(30)
    t1 = time.perf_counter()
    us = int((t1 - t0) * 1_000_000)
    print(f"W1_FIB checksum={res} time_us={us}")

if __name__ == "__main__":
    main()
